node/src/blockchain/blockchain_interface/lower_level_interface.rs

```diff
@@ -1,21 +1,66 @@
 // Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
-use crate::blockchain::blockchain_interface::data_structures::errors::BlockchainResult;
+use actix::Recipient;
+use crate::blockchain::blockchain_interface::data_structures::errors::{BlockchainError, PayableTransactionError};
 use crate::sub_lib::wallet::Wallet;
-use ethereum_types::U64;
-use web3::types::U256;
+use ethereum_types::{H256, U64};
+use futures::Future;
+use web3::contract::Contract;
+use web3::transports::Http;
+use web3::types::{Address, Filter, Log, TransactionReceipt, U256};
+use masq_lib::blockchains::chains::Chain;
+use masq_lib::logger::Logger;
+use crate::accountant::db_access_objects::payable_dao::PayableAccount;
+use crate::blockchain::blockchain_bridge::PendingPayableFingerprintSeeds;
+use crate::blockchain::blockchain_interface::blockchain_interface_web3::lower_level_interface_web3::TransactionReceiptResult;
+use crate::blockchain::blockchain_interface::data_structures::ProcessedPayableFallible;
 
 pub trait LowBlockchainInt {
-    fn get_transaction_fee_balance(&self, wallet: &Wallet) -> ResultForBalance;
+    // TODO: GH-495 The data structures in this trait are not generic, will need associated_type_defaults to implement it.
+    // see issue #29661 <https://github.com/rust-lang/rust/issues/29661> for more information
```

Comment:
I'm not opposed to this wholly, I just need to direct some inaccuracy and maybe even help you to change your mind if you find it appropriate. 

It's quite unclear why you call for "associated type defaults". At the moment there are definitely techniques that can be used to make some of the types from this interface generic, but only in case you really mean the same think which I think you do. Is this that you want the implementator of this trait to have to declare its own specific types taking part in the methods' signatures?  

Maybe you did think further and therefore touched a thought of reducing the amount of types that would have to be declared differently with each new implementation. In other words, maybe you thought that it would be nice to have a chance to specify your own types, most suitable for the particular implementation, but also it would be nice to have some sort of defaults and you could focus only on those that matter for you. If that's right, your comment's second part makes more sense, however, then you should make the comment and maybe even the description of the card speak more clearly of what you had in mind, what role the defaults play in it, and so why the waiting.

Reply:
We have added a comment to card GH-495: https://github.com/MASQ-Project/Node/issues/495#issuecomment-2426338738